lua/telescope/utils.lua

```diff
@@ -446,7 +446,9 @@ utils.transform_devicons = load_once(function()
         return display
       end
 
-      local icon, icon_highlight = devicons.get_icon(utils.path_tail(filename), nil, { default = true })
+      local basename = utils.path_tail(filename)
+      -- the double :e:e gets multipart extensions, if one exists, like *.test.js, for example
+      local icon, icon_highlight = devicons.get_icon(basename, vim.fn.fnamemodify(basename, ":e:e"), { default = true })
```

Comment:
can we not use `vim.fn.fnamemodify`? Using `vim.fn` here makes this function no longer api-fast and i prefer to keep it api-fast because entry-maker can run inside a tight lua loop and this function is primarily used there